std/container/array.d

```diff
@@ -318,7 +318,25 @@ if (!is(Unqual!T == bool))
                 const nbytes = mulu(newLength, T.sizeof, overflow);
                 if (overflow)
                     assert(0);
-                _payload = (cast(T*) realloc(_payload.ptr, nbytes))[0 .. newLength];
+
+                static if (hasIndirections!T)
+                {
+                    auto newPayloadPtr = cast(T*) malloc(nbytes);
+                    newPayloadPtr || assert(false, "std.container.Array.length failed to allocate memory.");
+                    auto newPayload = newPayloadPtr[0 .. newLength];
+                    memcpy(newPayload.ptr, _payload.ptr, _payload.sizeof);
```

Comment:
@rjframe this introduced a regression, [18800 – [REG2.080.0-beta.1] Array.length setter segfaults for payloads with indirections](https://issues.dlang.org/show_bug.cgi?id=18800).